opengrok-indexer/src/main/java/org/opengrok/indexer/history/GitRepository.java

```diff
@@ -171,14 +174,18 @@ Executor getRenamedFilesExecutor(final File file, String sinceRevision) throws I
     /**
      * Try to get file contents for given revision.
      *
+     * @param sink a required target sink
+     * @param outIterations a required out array for storing the number of calls
+     * made to {@code sink}
      * @param fullpath full pathname of the file
      * @param rev revision
-     * @return contents of the file in revision rev
+     * @return {@code true} if any contents were found
      */
-    private InputStream getHistoryRev(String fullpath, String rev) {
-        InputStream ret = null;
-        File directory = new File(getDirectoryName());
+    private boolean getHistoryRev(BufferSink sink, int[] outIterations,
```

Comment:
please dont use array for out param, think about something better, object reference or returning a pair or smth else